packages/sdk/src/oracle/types/interfaces.ts

```diff
@@ -1,17 +1,54 @@
 import { BigNumberish, BigNumber, Signer, TransactionResponse, TransactionReceipt, Provider } from "../types/ethers";
-import { RequestState, RequestKey, Request as RequestFromEvent } from "../../clients/optimisticOracleV2";
+import { RequestState, RequestKey } from "../../clients/optimisticOracle";
 import { Client } from "../client";
 import { OracleType } from "../types/state";
 
+// This object diverges from the v1 and v2 request shapes. This is to normalize
+// the data into a common shape. This has pitfalls and will be problematic if oracle implementations
+// diverge too much, but so far it seems ok
+export type Request = RequestKey & {
+  chainId: number;
+} & Partial<{
+    // this is partial since we dont know what events we have to populate parts of this
+    proposer: string;
+    disputer: string;
+    currency: string;
+    settled: boolean;
+    proposedPrice: BigNumber;
+    resolvedPrice: BigNumber;
+    expirationTime: BigNumber;
+    reward: BigNumber;
+    finalFee: BigNumber;
+    price: BigNumber;
+    payout: BigNumber;
+    state: RequestState;
+    // metadata about the transaction that triggered the state changes
+    requestTx: string;
+    proposeTx: string;
+    disputeTx: string;
+    settleTx: string;
+    requestBlockNumber: number;
+    proposeBlockNumber: number;
+    disputeBlockNumber: number;
+    settleBlockNumber: number;
+    // oo v2 fields moved here from settings object
+    bond: BigNumber;
+    customLiveness: BigNumber;
+    eventBased: boolean; // True if the request is set to be event-based.
+    refundOnDispute: boolean; // True if the requester should be refunded their reward on dispute.
+    callbackOnPriceProposed: boolean; // True if callbackOnPriceProposed callback is required.
+    callbackOnPriceDisputed: boolean; // True if callbackOnPriceDisputed callback is required.
+    callbackOnPriceSettled: boolean; // True if callbackOnPriceSettled callback is required.
```

Comment:
How is this set on OOv1 requests that don't have this parameter?

Reply:
the reason these are optional btw is because we cant guarantee we will see events/requests to populate the object, so we have to be robust to undefined data on the FE

Reply:
I think that seems reasonable.

Do _all_ of these fields need to be optional (for instance, the initialization params, like bond, currency, settled, etc)? I agree that these will sometimes need to be undefined, although it will imply something different about the field than for the other fields: this field doesn't exist for this OO rather than it not being specified yet.

Not a strong suggestion, but maybe a different way to manage fields that only exist for some types of OOs would be inherited types:
```ts
// all common fields
interface Request {
  ...
}

// Specific fields for a specific type of OO.
interface OOV2Request extends Request {
    callbackOnPriceProposed: boolean; // True if callbackOnPriceProposed callback is required.
    callbackOnPriceDisputed: boolean; // True if callbackOnPriceDisputed callback is required.
    callbackOnPriceSettled: boolean; // True if callbackOnPriceSettled callback is required.
}
```

You can store and construct the OOv2 types inside your OOv2 class, but you can always implicitly downcast and return them as normal Requests for common methods.

It was just something I was thinking about looking through the specific OO class implementations.